File: backend/pokerogue/src/main/java/com/pokerogue/helper/pokemon/data/Evolution.java
package com.pokerogue.helper.pokemon.data;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@AllArgsConstructor
@NoArgsConstructor
public final class Evolution {

    private String from;
    private String level;
    private String to;
    private String item;
    private String condition;

    public String from() { // Todo: 이 메소드들 지우기
        return from;
    }

    public String level() {
        return level;
    }

    public String to() {
        return to;
    }

    public String item() {
        return item;
    }

    public String condition() {
        return condition;
    }
}

File: android/data/src/main/java/poke/rogue/helper/data/model/Evolution.kt
package poke.rogue.helper.data.model

import poke.rogue.helper.remote.dto.response.pokemon.EvolutionResponse

data class Evolution(
    val pokemonId: String,
    val pokemonName: String,
    val level: Int,
    val item: String?,
    val condition: String?,
)

fun EvolutionResponse.toData(): Evolution =
    Evolution(
        pokemonId = pokemonId,
        pokemonName = pokemonName,
        level = level,
        item = item,
        condition = condition,
    )

fun List<EvolutionResponse>.toData(): List<Evolution> =
    map(EvolutionResponse::toData)